test/quantization/test_numeric_suite_fx.py

```diff
@@ -573,6 +573,34 @@ def test_matching_failure_node_type(self):
         with self.assertRaises(GraphMatchingException) as ex:
             results = get_matching_subgraph_pairs(mp1, mp2)
 
+    @override_qengines
+    def test_nodes_before_cat(self):
+        # verify that nodes before cat get matched
+        class M(nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x0):
+                x1 = F.relu(x0)
+                y1 = F.relu(x0)
+                x2 = torch.cat([x1, y1])
+                return x2
+
+        m = M().eval()
+        mp = prepare_fx(m, {'': torch.quantization.default_qconfig})
+        # TODO(future PR): prevent the need for copying here, we can copy the
+        # modules but should reuse the underlying tensors
+        mp_copy = copy.deepcopy(mp)
+        mq = convert_fx(mp_copy)
+        results = get_matching_subgraph_pairs(mp, mq)
+
+        expected_types = {
+            'cat_1': ((torch.cat, torch.cat), (toq.cat, toq.cat)),
```

Comment:
Why are there 2 cats here when the graph has a single cat node?

Reply:
tl;dr;
* we need to use subgraphs to properly support fusions
* the current definition of subgraph is (start_node, end_node) (this may change in the future for more complicated subgraphs)
* this code is for a subgraph of one node, with start_node being the same as end_node